### PR TITLE
Adds devcontainer and Autogenerated Docs

### PR DESCRIPTION
**Pull request by avb-is-me**
## Describe Your Changes

-Add .devcontainer starter json with node:20 so people can run on codespaces and devpod.  Electron does not work great on codespaces, but the server and other aspects run fine.

-Adds Autogenerated Docs for how to use the local API server.  All the Docs are AI generated without refinement and I added the _ in front of them so you can delete, keep, or remove the underscore if you want to show the documentation.  I am sure these docs are not perfect but would love to get your feedback on some of them, or any improvements I can make to help the team save time and energy writing docs from scratch.

### CODE REVIEW

File: .devcontainer/devcontainer.json
Comment (hieu-jan) [inline .devcontainer/devcontainer.json:1]: Thank you for this contribution. Development container will help to configure a consistent dev environment.

**Comment by avb-is-me** [comment]
@hieu-jan I'll try to divide into different pull requests

**Comment by avb-is-me** [comment]
@hieu-jan https://github.com/janhq/jan/pull/1739

**Comment by hieu-jan** [comment]
Hi @avb-is-me, I merged your separated PR of devcointainer.json. Thank you so much for your contribution. For API Reference, I also updated https://jan.ai/api-reference/ with the input and output of real example for current Jan server. Feel free to made any pull request to modify this page if you find any bugs.

**Pull request closed without merging**